feedback/filters.py

```diff
@@ -156,6 +169,76 @@ def filter(self, qs, value):
         return qs.filter_flags(*value)
 
 
+class ComboTextSearchWidget(forms.MultiWidget):
+    """Widget for displaying a text search field with a checkbox
+    that allows using another search method.
+    The checkbox label and helptext need to be defined through widget
+    attributes 'bool_label' and 'bool_helptext'.
+    """
+    template_name = "feedback/widgets/combo_textsearch_widget.html"
+
+    def __init__(self, attrs=None):
+        widgets = {
+            'text': forms.TextInput(attrs=attrs),
+            'regex': forms.CheckboxInput(attrs=attrs),
+        }
+        super().__init__(widgets, attrs)
+
+    def decompress(self, value) -> tuple[str, bool]:
+        if value is None:
+            return [None, None]
+        return value
+
+
+class ComboTextSearchField(forms.MultiValueField):
+    widget = ComboTextSearchWidget
+
+    def __init__(self, *args, **kwargs):
+        fields = (
+            forms.CharField(required=False),
+            forms.BooleanField(required=False),
+        )
+        kwargs.setdefault('require_all_fields', False)
+        super().__init__(fields, *args, **kwargs)
+
+    def compress(self, data_list: tuple[str, bool]) -> tuple[str, bool]:
+        return data_list
+
+    def validate(self, value: tuple[str, bool]):
+        """Validate field value, especially when regex search is used.
+        If an another search method is used as alternative instead of
+        regex or iregex, this method should be overwritten.
+        """
+        super().validate(value)
+        # if regex search is selected, check that string is valid regex
+        if value[1]:
+            validate_regex(value[0])
+
+
+class ComboTextSearchFilter(django_filters.CharFilter):
+    """Text search filter that supports two different search methods.
+    By default, the filter uses the filter_text method of the queryset.
+    However, if the checkbox is selected, uses alternative method for
+    searching (which is by default case-insensitive regex search).
+    """
+    field_class = ComboTextSearchField
+
+    def __init__(self, *args, lookup_expr=None, **kwargs):
```

Comment:
`__init__` should have a return type annotated with `-> None`:
https://peps.python.org/pep-0484/#the-meaning-of-annotations